apps/web/src/screens/Profile/modals/ConfirmTransaction/index.tsx

```diff
@@ -1,48 +1,96 @@
+import { FC, useEffect, useState } from 'react';
+import {
+	getWalletPublicKey,
+	modalActions,
+	ModalConfigs,
+	sendToken,
+} from '@walless/app';
+import { Networks } from '@walless/core';
 import { Stack } from '@walless/gui';
 
 import ModalWrapper from '../components/ModalWrapper';
 import {
-	networkLogo,
-	networkName,
-	tokenAmount,
-	tokenName,
+	// networkLogo,
+	// networkName,
+	// tokenAmount,
+	// tokenName,
 	walletAddress,
 	walletName,
 } from '../internal';
 import NavBtn from '../SendToken/components/NavBtn';
+import { DropdownItemProps } from '../SendToken/internal';
 
 import AccountInfo from './AccountInfo';
 import Header from './Header';
 import RecipientInfo from './RecipientInfo';
 
-const ConfirmTransactionScreen = () => {
+interface RequiredContext {
+	token: DropdownItemProps;
+	network: DropdownItemProps;
+	receiver: string;
+	amount: number;
+	parent: {
+		id: string;
+	};
+}
+
+const ConfirmTransactionScreen: FC<{
+	config: ModalConfigs & { context: RequiredContext };
+}> = ({ config }) => {
+	const { token, network, receiver, amount, parent } = config.context;
+
+	const handleOnPressGoBackBtn = () => {
+		modalActions.destroy(config.id);
+	};
+
+	const handleOnPressCloseBtn = () => {
+		modalActions.destroy(config.id);
+
+		if (parent) {
+			modalActions.destroy(parent.id);
+		}
+	};
+
+	const [address, setAddress] = useState<string | null>(null);
+
+	useEffect(() => {
+		(async () => {
+			setAddress((await getWalletPublicKey(network.value as Networks)) || null);
+		})();
+	}, []);
```

Comment:
instead of "fetch" public key in-place, store (pre-cache) it under Valtio state, and select from that existed state instead. Valtio use `IndexedDB` as public-key source.